### PR TITLE
Improved Code Autocompletion

### LINKED ISSUE #33425: Autocompletion doesn't suggest identifiers that begin differently
**Godot version:**
3.2 beta1

**Issue description:**
Basically this:
image

`p_attack` will only appear if I start the word with "p". It should be matched to "attack" too.

Screen from VS Code for reference
image

Slightly related to #21726

### PR DESCRIPTION
**Pull request by code-xD**
Fixes #33425 improved code suggestions. Although work needs to be done on the GUI side to make sure the right letters of suggestions are highlighted.

### CODE REVIEW

**Comment by YeldhamDev** [comment]
I accidentally carried over the other commit.

File: editor/quick_open.cpp [hunk 1]
```diff
@@ -113,12 +113,12 @@ void EditorQuickOpen::_sbox_input(const Ref<InputEvent> &p_ie) {
...
 
+	int pos = -1;
 	if (search == path) {
 		return 1.2f;
-	}
-	if (path.findn(search) != -1) {
-		return 1.1f;
-	}
+	} else if ((pos = path.rfindn(search)) != -1)
+		return 1.1f + 0.09 / (path.length() - pos + 1);
```
Comment (aaronfranke) [inline editor/quick_open.cpp]: Braces will be enforced in the future, see #33027

**Comment by code-xD** [comment]
@aaronfranke @YeldhamDev I made the required changes you can now re-review the code.

**Comment by bojidar-bg** [comment]
Doesn't this still break in cases like this?
```gdscript
class X:
    const xabcxy = 0
    const xaxbxcy = 1
func _ready():
    print(X.abc<tab>) # Should display both, since X.abcy<tab> displays both
```

**Comment by aaronfranke** [comment]
@bojidar-bg Why would you want that? I don't see the use case for matching non-contiguous characters. By that logic, my previous sentence would match with "donuts" and "contact".

Btw, the old code doesn't do that either, so this PR is only an improvement from what I see.

**Comment by bojidar-bg** [comment]
What I meant to say is that while it is definitely an improvement, it is still possible to reproduce a form of #33425 after it.

**Comment by akien-mga** [comment]
Superseded by #34288, but thanks for the contribution nevertheless!

**Pull request closed without merging**